mantidproject/mantid
# Add a new option to AppendSpectra to append Y-axis values

### Description of work
This PR adds a new `AppendYAxisLabels` option to the AppendSpectra algorithm. This feature allows for copying y-axis values from two input workspaces to the output workspace. This is particularly useful for workspaces like sqw, which use different types of axes, such as numeric and text. The main purpose of this update is to solve a problem with appending spectra from sqw workspaces in the Elwin tab.
#### Summary of work

Add a new option (AppendYAxisLabels) to `AppendSpectra` to append y-axis values from input workspaces to the output workspace.


Fixes #36299. 




#### Further detail of work

The issue was identified in this [line](https://github.com/mantidproject/mantid/blob/main/qt/scientific_interfaces/Inelastic/Manipulation/InelasticDataManipulationElwinTabModel.cpp#L110). When using workspace input tab, the spectra from the workspace are displayed in the table, and the user has the option to remove any of them. These spectra are then merged into a single workspace to be passed to the primary algorithm, “ElasticWindowMultiple”.
However, a problem arises when using SQW workspaces. The axes, which contain theta values necessary for computing Q values, are not copied. To address this, I’ve introduced a new parameter that enables the copying of axes from the input workspaces to the output workspace. I've chosen to make this a parameter because it is not straightforward how axes are being copied during the appending process. 

### To test:
1- Go to Interfaces->Inelastic->Data Manipulation->Elwin tab.
2- Browse to BIO92_sqw.nxs file. 
3- Click Run
2- Switch from the File tab to the Workspace tab.
3- Click Add Workspace then browse to the file BIO92_sqw.nxs then
4- Check All Spectra and click Add
5- Click Run
6- Compare the three output workspaces of the two methods and they should be the same
7- Remove some spectra from the table 
8- Click run and then check the output and it should has some q values not just q=0 for all values







---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
